Source/monster.cpp

```diff
@@ -1719,7 +1720,7 @@ void MonsterPetrified(Monster &monster)
 	}
 }
 
-Monster *AddSkeleton(Point position, Direction dir, bool inMap)
+std::optional<Monster *> AddSkeleton(Point position, Direction dir, bool inMap)

```

Comment:
It looks like you are both returning an optional and using a pointer, instead you should return an optional reference.

Reply:
I tried, but the compiler doesn't like it 😞 

https://stackoverflow.com/a/40383290/360829